Revise recent master failures

<!-- ⚠️ If you do not respect this template, your issue will be closed -->
<!-- ⚠️ Make sure to browse the opened and closed issues -->

### What is the current behavior?

During today's round of commits to `master`, a couple of items seem to have cause errors in the tests:

From https://dev.azure.com/qiskit-ci/qiskit-ibmq-provider/_build/results?buildId=1736:
* `TestIBMQJob.test_custom_job_name` and `TestIBMQJob.test_duplicate_job_name` are failing when asserting over the lengths:
  ```
  >       self.assertEqual(len(job_list), 1)
  E       AssertionError: 0 != 1
  ```
  They are likely a result of the tests attempting to be run against the old API, where we don't support job names - if that is the case, can we revise the tests so they are only executed in the new API?

From https://travis-ci.com/Qiskit/qiskit-ibmq-provider/jobs/224723566:
* one test seems to have failed:
```
ERROR: test_online_qasm_simulator_two_registers (test.ibmq.test_ibmq_qasm_simulator.TestIbmqQasmSimulator)
Test online_qasm_simulator_two_registers.
----------------------------------------------------------------------
Traceback (most recent call last):
  File "/home/travis/build/Qiskit/qiskit-ibmq-provider/test/decorators.py", line 114, in _wrapper
    return func(obj, *args, **kwargs)
  File "/home/travis/build/Qiskit/qiskit-ibmq-provider/test/decorators.py", line 141, in _wrapper
    return func(*args, **kwargs)
  File "/home/travis/build/Qiskit/qiskit-ibmq-provider/test/ibmq/test_ibmq_qasm_simulator.py", line 101, in test_online_qasm_simulator_two_registers
    result = job.result()
  File "/home/travis/build/Qiskit/qiskit-ibmq-provider/qiskit/providers/ibmq/job/ibmqjob.py", line 255, in result
    'it is {}'.format(str(status)))
qiskit.providers.exceptions.JobError: 'Invalid job state. The job should be DONE but it is JobStatus.ERROR'
```

This one might be a one-timer, but can we double-check?

--- a/test/ibmq/test_ibmq_job.py
+++ b/test/ibmq/test_ibmq_job.py
@@ -31,7 +31,7 @@ from qiskit.test import slow_test
 from qiskit.compiler import assemble, transpile
 
 from ..jobtestcase import JobTestCase
-from ..decorators import requires_provider, requires_qe_access
+from ..decorators import requires_provider, requires_qe_access, requires_new_api_auth
 
 
 class TestIBMQJob(JobTestCase):
@@ -433,9 +433,13 @@ class TestIBMQJob(JobTestCase):
         job = backend.run(qobj)
         _ = job.result()
 
-    @requires_provider
-    def test_custom_job_name(self, provider):
+    @requires_qe_access
+    @requires_new_api_auth
+    def test_custom_job_name(self, qe_token, qe_url):
         """Test assigning a custom job name."""
+        factory = IBMQFactory()
+        provider = factory.enable_account(qe_token, qe_url)
+
         backend = provider.get_backend('ibmq_qasm_simulator')
         qobj = assemble(transpile(self._qc, backend=backend), backend=backend)
 
@@ -447,9 +451,13 @@ class TestIBMQJob(JobTestCase):
         self.assertEqual(len(job_list), 1)
         self.assertEqual(job_id, job_list[0].job_id())
 
-    @requires_provider
-    def test_duplicate_job_name(self, provider):
+    @requires_qe_access
+    @requires_new_api_auth
+    def test_duplicate_job_name(self, qe_token, qe_url):
         """Test multiple jobs with the same custom job name."""
+        factory = IBMQFactory()
+        provider = factory.enable_account(qe_token, qe_url)
+
         backend = provider.get_backend('ibmq_qasm_simulator')
         qobj = assemble(transpile(self._qc, backend=backend), backend=backend)
 

